core/domain/stats_services_test.py

```diff
@@ -1151,6 +1159,223 @@ def test_cst_exp_issue_is_updated_when_state_is_renamed(self):
             actions[3]['action_customization_args']['state_name']['value'],
             'Z')
 
+    def test_migrate_to_latest_action_schema_riases_error_from_v0(self):
+        stats_services.save_exp_issues_model(
+            stats_domain.ExplorationIssues(self.exp.id, self.exp.version, [
+                self._create_cst_exp_issue(
+                    [self._create_cst_playthrough(['A', 'B', 'A'])],
+                    ['A', 'B', 'A'])
+            ]))
+
+        exp_issues = (
+            stats_services.get_exp_issues(self.exp.id, self.exp.version))
+
+        exp_issue = exp_issues.unresolved_issues[0]
+
+        stats_models.CURRENT_ACTION_SCHEMA_VERSION = 0
+        model = stats_models.PlaythroughModel.get_by_id(
+            exp_issue.playthrough_ids[0])
+
+        with self.assertRaisesRegex(
+            Exception, 'Sorry, we can only process '
+            'v1-v%d and unversioned action schemas '
+            'at present.' % stats_models.CURRENT_ACTION_SCHEMA_VERSION):
```

Comment:
To avoid having the contents of the `with` block at the same level of indentation as the arguments to `with`
```suggestion
        with self.assertRaisesRegex(
            Exception, 'Sorry, we can only process '
            'v1-v%d and unversioned action schemas '
            'at present.' % stats_models.CURRENT_ACTION_SCHEMA_VERSION
        ):
```

ditto elsewhere

Reply:
not done